pkg/api/imagereferencemutators/pods.go

```diff
@@ -94,36 +96,66 @@ func GetPodSpec(obj runtime.Object) (*kapi.PodSpec, *field.Path, error) {
 // This only returns pod specs for v1 compatible objects.
 func GetPodSpecV1(obj runtime.Object) (*kapiv1.PodSpec, *field.Path, error) {
 	switch r := obj.(type) {
+
 	case *kapiv1.Pod:
 		return &r.Spec, field.NewPath("spec"), nil
+
 	case *kapiv1.PodTemplate:
 		return &r.Template.Spec, field.NewPath("template", "spec"), nil
+
 	case *kapiv1.ReplicationController:
 		if r.Spec.Template != nil {
 			return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
 		}
+
 	case *extensionsv1beta1.DaemonSet:
 		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1.DaemonSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1beta2.DaemonSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+
 	case *extensionsv1beta1.Deployment:
 		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1beta1.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1beta2.Deployment:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+
 	case *extensionsv1beta1.ReplicaSet:
 		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1.ReplicaSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+	case *appsv1beta2.ReplicaSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+
 	case *batchv1.Job:
 		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
+
 	case *batchv2alpha1.CronJob:
 		return &r.Spec.JobTemplate.Spec.Template.Spec, field.NewPath("spec", "jobTemplate", "spec", "template", "spec"), nil
+
 	case *batchv2alpha1.JobTemplate:
 		return &r.Template.Spec.Template.Spec, field.NewPath("template", "spec", "template", "spec"), nil
+
+	case *appsv1.StatefulSet:
+		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
 	case *appsv1beta1.StatefulSet:
 		return &r.Spec.Template.Spec, field.NewPath("spec", "template", "spec"), nil
-	case *appsv1beta1.Deployment:
+	case *appsv1beta2.StatefulSet:
```

Comment:
is this the only change here? can you pls minimize the changes?

Reply:
It is minimal, but required in the long run.